chispa/dataframe_comparer.py

```diff
@@ -22,25 +30,7 @@ def assert_df_equality(df1, df2, ignore_nullable = False, transforms = [], allow
     rows1 = df1.collect()
     rows2 = df2.collect()
     if allow_nan_equality:
-        zipped = list(six.moves.zip_longest(rows1, rows2))
-        t = PrettyTable(["df1", "df2"])
-        allRowsEqual = True
-        for r1, r2 in zipped:
-            # rows are not equal when one is None and the other isn't
-            if (r1 is not None and r2 is None) or (r2 is not None and r1 is None):
-                allRowsEqual = False
-                t.add_row([r1, r2])
-            # rows are equal
-            elif are_rows_equal_enhanced(r1, r2, True):
-                first = bcolors.LightBlue + str(r1) + bcolors.LightRed
-                second = bcolors.LightBlue + str(r2) + bcolors.LightRed
-                t.add_row([first, second])
-            # otherwise, rows aren't equal
-            else:
-                allRowsEqual = False
-                t.add_row([r1, r2])
-        if allRowsEqual == False:
-            raise DataFramesNotEqualError("\n" + t.get_string())
+        assert_generic_df_equality(df1, df2, are_rows_equal_enhanced, [True])
     else:
```

Comment:
Are you planning to also modify this `else` block to call one of the other utility functions you have? e.g. Call `assert_generic_df_equality()` but with slightly different arguments so it doesn't consider NaNs equal.

Reply:
Good catch, created a separate `assert_basic_df_equality` method to make the intentions of the code clearer.  What I want to do now is benchmark `assert_basic_df_equality` vs `assert_generic_df_equality(df1, df2, are_rows_equal, [])` to see which is faster.  Both should give the same response and want to give users the fastest option (cause slow Spark tests are painful).